futurex_openedx_extensions/helpers/apps.py

```diff
@@ -29,8 +30,19 @@ class HelpersConfig(AppConfig):
 
     def ready(self) -> None:
         """Connect handlers to send notifications about discussions."""
+        from lms.djangoapps.course_home_api.permissions import (  # pylint: disable=import-outside-toplevel
+            CAN_MASQUARADE_LEARNER_PROGRESS,
+        )
+        from lms.djangoapps.courseware.rules import (  # pylint: disable=import-outside-toplevel
+            HasAccessRule,
+            HasRolesRule,
+        )
+
         from futurex_openedx_extensions.helpers import \
             custom_roles  # pylint: disable=unused-import, import-outside-toplevel
         from futurex_openedx_extensions.helpers import \
             monkey_patches  # pylint: disable=unused-import, import-outside-toplevel
         from futurex_openedx_extensions.helpers import signals  # pylint: disable=unused-import, import-outside-toplevel
+        if CAN_MASQUARADE_LEARNER_PROGRESS in perms:
+            del perms[CAN_MASQUARADE_LEARNER_PROGRESS]
```

Comment:
I don't think this line is needed. I'm presuming `perms` is a dictionary that we can set to without deleting, right?

Reply:
Without this it was giving error that permission already existed.
![image](https://github.com/user-attachments/assets/a9049900-d8f2-462d-8ba1-59185e033e89)

Reply:
Without this it was giving error that permission already existed.
![image](https://github.com/user-attachments/assets/a9049900-d8f2-462d-8ba1-59185e033e89)

Reply:
yes, this is explicitly prevented in `PermissionMap`. I searched and I found that there was no technical reason behind it. It's just to prevent accidental change of the permission. So, a `del` statement is needed which will alarm reviewers as it did here!

Reply:
Makes sense, thank you both!!